chart/templates/_helpers.yaml

```diff
@@ -197,6 +197,7 @@ If release name contains chart name it will be used as a full name.
   imagePullPolicy: {{ .Values.images.gitSync.pullPolicy }}
   securityContext: {{- include "localContainerSecurityContext" .Values.dags.gitSync | nindent 4 }}
   env:
+    {{- if semverCompare "<4.0.0" .Values.images.gitSync.tag }}
```

Comment:
I don't think we can just compare against the tag like this. Sure, works for the upstream images, but we shouldn't restrict how folks can retag in their own registry.

If we want to do branching logic like this, we'll have to add an equivalent to `airflowVersion` somewhere.

Reply:
Alternatively, can we just set both the old and new env vars? A little messier, but no new version config either...

Reply:
Moving @hussein-awala's question here from a resolved thread, I'd say that will dictate whether the old + new approach is palatable:

> Will we have deprecation warning if we provide both old and new styles to git-sync v4.0.0+ image?

Reply:
yes, we will have deprecation warning if we provide both old and new env vars